src/playground/services/invoices/delete/endpoint.clj

```diff
@@ -3,22 +3,29 @@
    [clojure.spec.alpha :as spec]
    [clojure.java.jdbc :as jdbc]
    [honeysql.core :as h]
+   [buddy.auth :refer [authenticated? throw-unauthorized]]
+   [io.pedestal.http.route :refer (url-for)]
    [playground.services.invoices.delete.logic :as logic]
    [playground.services.invoices.retrieve-all.logic :as retrieve-all.logic]
-   [playground.models.user :as models.user]))
+   [playground.models.user :as models.user]
+   [ring.util.response :as ring-resp]))
 
 (spec/def ::api (spec/keys :req-un [::models.user/id]))
 
-(defn perform [{{:keys [id]} :path-params :keys [db] :as request}]
-  (let [db (->> db :pool (hash-map :datasource))
-        _  (->> (logic/to-delete id)
-                (h/format)
-                (jdbc/execute! db))
-        fetch  (h/format (retrieve-all.logic/query-all))
-        left   (->> (logic/to-serialize)
+(defn perform [{{:keys [id]} :path-params :keys [db session] :as request}]
+  (let  [role (get-in session [:identity :role])]
+    (if (= "admin" role)
```

Comment:
`"admin"` can be considered a magic string, hardcoded value, etc.

Instead, do `(def admin-role "admin")` in playground.models.user.